messaging/src/main/java/org/axonframework/eventhandling/pooled/WorkPackage.java

```diff
@@ -257,6 +314,24 @@ private void processEvents() throws Exception {
         }
     }
 
+    private void consumeEntry(List<TrackedEventMessage<?>> eventBatch) {
+        ProcessingEntry entry = processingQueue.poll();
+        assert entry != null;
```

Comment:
The use of the `assert` keyword is tricky. It may be disabled at runtime. If we think `null` is a viable result (which in this scope it is, but outside the method you see that there is a non-empty check on the queue), then we should just do a null-check and perhaps just silently exit the method?

Reply:
I wonder how I let this in. Definitely wasn't my intention. I will remove this line.